src/setupWorker/stop/createStop.ts

```diff
@@ -1,6 +1,9 @@
-import { SetupWorkerInternalContext } from '../glossary'
+import { SetupWorkerInternalContext, StartOptions } from '../glossary'
 
-export const createStop = (context: SetupWorkerInternalContext) => {
+export const createStop = (
+  context: SetupWorkerInternalContext,
+  options?: StartOptions,
```

Comment:
It appears we don't have access to start options in the `createStop` function:

They are both created with the same internal `context` as an argument:

https://github.com/mswjs/msw/blob/8a5f573aed97dadce448df7e4acfb0d23d8eb18c/src/setupWorker/setupWorker.ts#L99-L100

However, the `createStart` later resolves its options

https://github.com/mswjs/msw/blob/8a5f573aed97dadce448df7e4acfb0d23d8eb18c/src/setupWorker/start/createStart.ts#L30-L31

and passed them to the function that attached the activation listener (and outputs the "Mocking enabled" message):

https://github.com/mswjs/msw/blob/8a5f573aed97dadce448df7e4acfb0d23d8eb18c/src/setupWorker/start/createStart.ts#L120-L123

https://github.com/mswjs/msw/blob/8a5f573aed97dadce448df7e4acfb0d23d8eb18c/src/setupWorker/start/utils/activateMocking.ts#L3-L10

In the case of `createStop`, we don't know anything about the start options. Wonder if we should persist them in context in any way 🤔

Reply:
@kettanaito Seems to be implemented in two different ways. Would we need a disableMocking util for this? :thinking:

Reply:
We don't have to break it into separate functions, as `createStop` does only a few things. The challenge is to reference the start options in this `createStop` function. Currently, there's no way to do that. What comes to my mind is to store those options in the `context` object available for these internal methods, but I bet it needs some more thinking.

Reply:
@kettanaito Do you mean storing them, and then using it like `context.options`, also with more thinking, do you mean as in decision making or it would take more time?

If storing options in `context`, it would require some refactoring as well I believe.

Reply:
Yes, that's one of the options. We need to be careful with how we store the start options in the context because the context can be accessed while the worker hasn't been started yet. In other words, the info we're about to store is set asynchronously and that should be evident when we're trying to access it. 

My initial thought was to create a getter function that returns a start options object or `null` if it's not defined yet. Not sure if that's the best way to go. 

@timdeschryver, what would you recommend in this situation?

Reply:
Alright! I also get notified for issues and PRs since I have enabled watch on MSW haha :wink: :raised_hands: :muscle: :smiling_face_with_three_hearts:

Reply:
I think there are two options here, and I think the solution @kettanaito suggested is the best one.
Since the options are initialized in `start`, we have no other choice than adding it to `context`.

Another option would be to move the options from `start` to `createServer`, that way the options can be passed down to `start` and `stop` but this is a breaking change. I also think the DX is better when the user uses `start` to provide the options (otherwise, the user will have to provide the options twice if they're using the browser and the node environment).